src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -440,7 +459,11 @@ public Void caseStringType() {
 
         @Override
         public Void caseObjectType(ObjectType type) {
-          emit("Object");
+          if (type.isRecordType()) {
```

Comment:
nope, I guess this is something closure added on top of rhino's very old type system?